### ISSUE TITLE
issue parsing config from setup.cfg

### ISSUE THREAD

**Issue by jeremyguarini** [hhatto/autopep8#306]
Hi,

I'm trying to define configuration values in my project's setup.cfg. Not having a real concrete example to go off of I tried the following:

[pep8]
aggressive = 1
experimental = 1
max-line-length = 165
verbose = 1

which mostly worked

I did get the following error when using python3

>(test_app) koshka:test_app jeremy$ autopep8  test.py
[file:test.py]
--->  Applying global fix for E265
--->  Applying global fix for W602
--->  2 issue(s) to fix {'W291': {5}, 'E226': {28}}
Traceback (most recent call last):
  File "/Users/jeremy/Dev/test_app/bin/autopep8", line 11, in <module>
    sys.exit(main())
  File "/Users/jeremy/Dev/test_app/lib/python3.5/site-packages/autopep8.py", line 3650, in main
    fix_multiple_files(args.files, args, sys.stdout)
  File "/Users/jeremy/Dev/test_app/lib/python3.5/site-packages/autopep8.py", line 3562, in fix_multiple_files
    _fix_file((name, options, output))
  File "/Users/jeremy/Dev/test_app/lib/python3.5/site-packages/autopep8.py", line 3543, in _fix_file
    fix_file(*parameters)
  File "/Users/jeremy/Dev/test_app/lib/python3.5/site-packages/autopep8.py", line 2968, in fix_file
    fixed_source = fix_lines(fixed_source, options, filename=filename)
  File "/Users/jeremy/Dev/test_app/lib/python3.5/site-packages/autopep8.py", line 2948, in fix_lines
    fixed_source = fix.fix()
  File "/Users/jeremy/Dev/test_app/lib/python3.5/site-packages/autopep8.py", line 553, in fix
    aggressive=self.options.aggressive))
  File "/Users/jeremy/Dev/test_app/lib/python3.5/site-packages/autopep8.py", line 469, in _fix_source
    for result in sorted(results, key=_priority_key):
  File "/Users/jeremy/Dev/test_app/lib/python3.5/site-packages/autopep8.py", line 2721, in filter_results
    if aggressive <= 0:
TypeError: unorderable types: str() <= int()

it seems SafeConfigParser() brings in all values as strings
see https://pymotw.com/2/ConfigParser/

> SafeConfigParser does not make any attempt to understand the option type. The application is expected to use the correct method to fetch the value as the desired type. get() always returns a string. Use getint() for integers, getfloat() for floating point numbers, and getboolean() for boolean values.

So with that in mind and totally not knowing if this is entirely expected I added the following in def read_config(args, parser) near the bottom:

```  
        defaults = dict((k.lstrip('-').replace('-', '_'), v)
                        for k, v in config.items('pep8'))
        for k, v in defaults.items():
            try:
                defaults[k] = int(v)
            except ValueError:
                pass
        parser.set_defaults(**defaults)
    except Error:
        # Ignore for now.
        pass

    return parser
```

now this "fixes" this issue for my but I'm sure will break something else.

thoughts?

**Comment by myint** [comment]
I think `pycodestyle` does this properly:

https://github.com/PyCQA/pycodestyle/blob/ca5350195031c8e29c50292404eaec848bbb073b/pycodestyle.py#L2208

**Comment by hhatto** [comment]
Thank you for reporting.

fix in 43f5eeb.
